storage/remote/queue_manager.go

```diff
@@ -575,11 +576,44 @@ func (t *QueueManager) sendMetadataWithBackoff(ctx context.Context, metadata []p
 	return nil
 }
 
+func isSampleOld(sampleAgeLimit time.Duration, ts int64) bool {
+	if sampleAgeLimit == 0 {
+		// If sampleAgeLimit is unset, then we never skip samples due to their age.
+		return false
+	}
+	limitTs := time.Now().Add(-sampleAgeLimit)
```

Comment:
`time.Now()` is potentially expensive; recommend not calling it multiple times like this.

Reply:
we have to get the current time every time we run `isSampleOld`. How do you suggest to improve this?

Reply:
Pass it in from the caller.  Or the caller's caller depending.

Reply:
yep, could name it `baseTime` and pass it from `isTimeSeriesOldFilter`